### PR TITLE
catalog-react: deprecate catalogRouteRef and deduplicate entityRouteRef

### PR DESCRIPTION
**Pull request by Rugvip**
This removes the usage of the `catalogRouteRef` from `@backstage/catalog-react`, so that we can deprecate it.

It also adds a page on the microsite for stable short-links, in this case `https://backstage.io/links?bind-routes`.

And some validation to help integrators catch errors in the app setup earlier by checking that there are no required external routes that haven't been bound.

### CODE REVIEW

File: packages/core-app-api/src/routing/validation.ts [hunk 1]
```diff
@@ -54,3 +61,30 @@ export function validateRoutes(
...
+      plugin.externalRoutes,
+    )) {
+      if (externalRouteRef.optional) {
+        continue;
+      }
+
+      if (!routeBindings.has(externalRouteRef)) {
+        throw new Error(
+          `External route '${name}' of the '${plugin.getId()}' plugin must be bound to a target route. ` +
+            'See https://backstage.io/link?bind-routes for details.',
```
[the remark this answers is not in the document]
Comment (Rugvip) [reply packages/core-app-api/src/routing/validation.ts:85]: Yep, just didn't wanna build that system just yet

File: plugins/catalog-react/src/routes.ts [hunk 2]
```diff
@@ -16,20 +16,43 @@
...
+/**
+ * A stable route ref that points to the catalog page for an individual entity.
+ *
+ * This `RouteRef` can be imported and used directly, and does not need to be referenced
+ * via an `ExternalRouteRef`.
+ *
+ * If you want to replace the `EntityPage` from `@backstage/catalog-plugin` in your app,
+ * you need to use the `entityRouteRef` as the mount point instead of your own.
+ */
+export const entityRouteRef = getOrCreateGlobalSingleton(
```
Comment (freben) [inline plugins/catalog-react/src/routes.ts:43]: interesting 🤔

Comment (freben) [reply plugins/catalog-react/src/routes.ts:43]: So this one is really meant to stay here, and not go into `plugin-catalog` I presume.

**Review by freben — APPROVED** [review]

File: plugins/catalog-react/src/routes.ts (hunk 2 above)
Comment (Rugvip) [reply plugins/catalog-react/src/routes.ts:43]: Yep, idd

**Pull request merged**